aospy/run.py

```diff
@@ -1,51 +1,68 @@
-"""`Run` class; for storing attributes of a model run or obs product."""
+"""Functionality for storing attributes of a model run or obs product."""
+from .timedate import ensure_datetime
 
-from .timedate import TimeManager
+
+def _datetime_or_default(date, default):
+    """Return a datetime.datetime object or a default.
+
+    Parameters
+    ----------
+    date : `None` or datetime-like object
+    default : The value to return if `date` is `None`
+
+    Returns
+    -------
+    `default` if `date` is `None`, otherwise returns the result of
+    `ensure_datetime(date)`
+
+    """
+    if date is None:
+        return default
+    else:
+        return ensure_datetime(date)
```

Comment:
This is what I meant in terms of special functionality for these type checks.  So maybe this goes in timedate (or utils/time) and then Model and Proj call it too?  The alternative is a separate class, c.f. my previous comment.